packages/cocoindex/hooks/proxy_manager.py

```
@@ -211,17 +494,19 @@ def cleanup_orphan(config: dict, project_dir: str) -> None:
 
 
 def _derive_port(project_dir: str, base_port: int, port_range: int) -> int:
-    """project_dir のハッシュから決定論的にポートを導出する。
-
-    port_range が 0 以下の場合は base_port をそのまま返す（固定モード）。
-    """
+    """project_dir のハッシュから決定論的にポートを導出する。"""
     if port_range <= 0:
         return base_port
     h = hashlib.md5(project_dir.encode()).hexdigest()
```

Ruff (0.15.10)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)